src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -1460,30 +1570,78 @@ void doDocument(DocumentManagement documentManagement,
             new Object[] { opentextDocId, fileDataSize });
         return;
     }
-    long versionNumber = version.getNumber();
-    String contextId = documentManagement.getVersionContentsContext(
-        documentNode.getID(), versionNumber);
-    ContentService contentService =
-        this.soapFactory.newContentService(documentManagement);
-    DataHandler dataHandler = contentService.downloadContent(contextId);
-
     String contentType = version.getMimeType();
     if (contentType != null) {
       response.setContentType(contentType);
     }
-    InputStream inputStream = dataHandler.getInputStream();
-    try {
-      if (inputStream != null) {
-        IOHelper.copyStream(inputStream, response.getOutputStream());
+
+    switch (this.downloadMethod) {
+      case "contentserver": {
+        String urlString = this.indexingContentServerUrl
+            + "?func=ll&objAction=download&objId=" + documentNode.getID();
+        log.log(Level.FINEST, "Downloading content from {0}",
+            urlString);
+        sendUrlResponse(response, urlString,
+            this.soapFactory.getAuthenticationToken(documentManagement));
+        break;
       }
-    } finally {
-      if (inputStream != null) {
+
+      case "cws": {
+        log.log(Level.FINEST, "Downloading content from CWS");
+        long versionNumber = version.getNumber();
+        String contextId = documentManagement.getVersionContentsContext(
+            documentNode.getID(), versionNumber);
+        ContentService contentService =
+            this.soapFactory.newContentService(documentManagement);
+        DataHandler dataHandler = contentService.downloadContent(contextId);
+        InputStream inputStream = dataHandler.getInputStream();
```

Comment:
I know this is all existing code, but unless diff -w shows it unchanged, we could switch to using try-with-resources here (and elsewhere).